src/domain/products/edit/sections/thumbnail/thumbnail-modal.tsx

```diff
@@ -0,0 +1,226 @@
+import { Product } from "@medusajs/medusa"
+import React, { useEffect } from "react"
+import {
+  Control,
+  FieldArrayWithId,
+  useFieldArray,
+  useForm,
+} from "react-hook-form"
+import FileUploadField from "../../../../../components/atoms/file-upload-field"
+import Button from "../../../../../components/fundamentals/button"
+import TrashIcon from "../../../../../components/fundamentals/icons/trash-icon"
+import Actionables, {
+  ActionType,
+} from "../../../../../components/molecules/actionables"
+import Modal from "../../../../../components/molecules/modal"
+import useNotification from "../../../../../hooks/use-notification"
+import { FormImage } from "../../../../../types/shared"
+import { prepareImages } from "../../../product-form/utils/helpers"
+import useEditProductActions from "../../hooks/use-edit-product-actions"
+
+type Props = {
+  product: Product
+  open: boolean
+  onClose: () => void
+}
+
+type ThumbnailForm = {
+  images: FormImage[]
+}
+
+const ThumbnailModal = ({ product, open, onClose }: Props) => {
+  const { onUpdate, updating } = useEditProductActions(product.id)
+  const {
+    control,
+    formState: { isDirty },
+    handleSubmit,
+    reset,
+  } = useForm<ThumbnailForm>({
+    defaultValues: getDefaultValues(product),
+  })
+  const { fields, remove, replace } = useFieldArray({
+    control: control,
+    name: "images",
+  })
+
+  const notification = useNotification()
+
+  useEffect(() => {
+    reset(getDefaultValues(product))
+  }, [product])
+
+  const onReset = () => {
+    reset(getDefaultValues(product))
+    onClose()
+  }
+
+  const onSubmit = handleSubmit(async (data) => {
+    let prepedImages: FormImage[] = []
```

Comment:
nit: spelling, "preppedImages" or "preparedImages"
```suggestion
    let preparedImages: FormImage[] = []
```